### PR TITLE
Expands Valhalla With A Vehicle Area, Adds Assault Vehicle Category to Valhalla Req Vendor

### PR DESCRIPTION
**Pull request by Wisemonster**
## About The Pull Request
Expands Valhalla with an area for testing vehicles (currently only the tank), and adds a vehicle ammo category to the valhalla req vendor, currently only contains the tank ammo. Requires the tank elevators/vendor to be changed to allow multiple vehicles across z-levels before it can be added to valhalla.
tgmc  _maps_map_files_generic_Admin_Level dmm  - StrongDMM 4_20_2024 7_55_50 PM

## Why It's Good For The Game
Prepares Valhalla for testing Assault vehicles whenever they are fully merged.

## Changelog

:cl:
add: Expanded west side of Valhalla with an assault vehicle area
add: Added an Assault Vehicle category to the Valhalla requisition vendor; added tank ammo to said category
/:cl:

### CODE REVIEW

**Comment by Wisemonster** [comment]
Note: the picture was before I added a bunch of cosmetic oil splatters around the assault vehicle area.

**Comment by Wisemonster** [comment]
Ok, it looks like I might have to replace the lattices next to the rwalls on the z-level, cause otherwise it's runtiming

**Comment by Wisemonster** [comment]
Ok, turns out what the linter means by neighbor is something on the same tile, so I just gotta fix some lattices on the same tiles as the walls

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 1]
```diff
@@ -1505,6 +1505,7 @@
 			/obj/structure/largecrate/supply/weapons/heavy_flakgun = -1,
 			/obj/item/ammo_magazine/heavy_isg/he = -1,
 			/obj/item/ammo_magazine/heavy_isg/sabot = -1,
+
 		),
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/marine_vending.dm:1509]:
```suggestion
		),
```

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 2]
```diff
@@ -1537,6 +1538,11 @@
 			/obj/item/implanter/chem/blood = -1,
 			/obj/item/implanter/blade = -1,
 		),
+		"Assualt Vehicle" = list(
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/marine_vending.dm:1541]:
```suggestion
		"Assault Vehicle" = list(
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Tank vendor currently only supports one vend of assault/non assault and this is shared globally

**Comment by Wisemonster** [comment]
Alright. I could place a tank in the area as a sort-of quick fix, or I could wait until multiple assault vehicles are spawnable then update the area with the tank vendor.

**Comment by Wisemonster** [comment]
> Tank vendor currently only supports one vend of assault/non assault and this is shared globally

To clarify, you mean the tank vendor connected to the elevator, not the tank parts vendor, right?

**Comment by TiviPlus** [comment]
> > Tank vendor currently only supports one vend of assault/non assault and this is shared globally
> 
> To clarify, you mean the tank vendor connected to the elevator, not the tank parts vendor, right?

Uhh wdym they're the same thing, but theres a globally tracked "this vehicle has been vended"

**Comment by TiviPlus** [comment]
You can do the place a few tanks workaround as a lazy fix and people blowing them up can cope but it should be changed eventualy

**Comment by Wisemonster** [comment]
Alright, updated pr to add a couple of tanks to the area, along with the parts for the tanks' primary and secondary slots. Removed the fabricator cause it's not going to do anything except maybe cause problems for living assault/transport crewmen till multiple vehicle elevators can exist.

**Review by TiviPlus — APPROVED** [review]

**Comment by TiviPlus** [comment]
@RipGrayson

**Review by RipGrayson — CHANGES_REQUESTED** [review]
Please move this carp somewhere else, two tiles away from any walls/windows or Valhalla's interior
carp7
You have space tiles here
carp8

**Comment by Wisemonster** [comment]
Alright, moved the carp and replaced the space tiles

**Pull request merged**